### ISSUE TITLE
unrecognizable characters with Qt document

### ISSUE THREAD

**Issue by magicdmer** [zealdocs/zeal#816]
when i look up qt5 document,  zeal display unrecognizable characters like follow picture , other document is ok

[problem picture](http://imgbox.com/OnVDi3M8)

**Comment by trollixx** [comment]
What's the language of your system? Also version of Zeal?

**Comment by magicdmer** [comment]
system language:  chinese
version: 0.4.0
It looks like that , the problem doesn't always happen but sometimes.

**Comment by trollixx** [comment]
Even for the same Qt 5 docset it is not always like that?

@annulen Any hints on how to debug this?

I can see that in a few places (search box, tabs) regular font somehow got replaced with a monospaced version, but at the same time menu and lists look fine.

**Comment by magicdmer** [comment]
Yes, even for the same Qt 5 docset,  it is not always like that.
The first time i reply to you,  it's ok, but now i reply to you , it's bad.

**Comment by annulen** [comment]
Have you tried to open the same html page in minimal QtWebKit application, e.g. https://github.com/OtterBrowser/test-qtwebkit ?

**Comment by annulen** [comment]
It's also interesting to know if this issue reproduces with Qt 5.4. Unfortunately it means building QtWebKit from sources

**Comment by trollixx** [comment]
@magicdmer Could you try viewing same pages with the linked minimal browser?

Zeal doesn't support Qt < 5.5.

I think Qt has issues with resolving and rendering fonts on Chinese Windows.

**Comment by annulen** [comment]
Is the page using local or web font? In online version I see it uses OpenSans, which is probably missing on Windows and used as a web font. Does rendering change if you install OpenSans into the system?

(Please check what font family is actually used)

**Comment by annulen** [comment]
Note that link above contains sources; if something binary is needed, you can try with Otter Browser

**Comment by trollixx** [comment]
`Titillium Web` is the font in use. It is a web font.

**Comment by annulen** [comment]
Indeed. Could you try painting some text with that font by modifying https://github.com/annulen/test_qrawfont and see if problem reproduces?

**Comment by trollixx** [comment]
It seems that Chinese Windows is more than just a language pack anyone could install...

**Comment by annulen** [comment]
This is quite possible. My current theory is a regression in QRawFont (btw there was one fixed recently, se https://bugreports.qt.io/browse/QTBUG-59799) which easily went unnoticed because it doesn't reproduce on vanilla system.

**Comment by annulen** [comment]
So, please try to find out if QRawFont + font file reproduces the bug or not

**Comment by annulen** [comment]
Did it ever happen for other system than Chinese Windows?

**Comment by trollixx** [comment]
I am not aware of this happening anywhere but the Chinese version.

**Comment by magicdmer** [comment]
But, other documents are ok except qt 5 document , other documents are never wrong.  and this once,  my qt5 document is back to normal. Look at picture:
[picture4](http://imgbox.com/rHLS10QQ)

**Comment by magicdmer** [comment]
In additon， when qt5 document is wrong, i use "Open URL" in chrome and it will be ok

**Comment by annulen** [comment]
@magicdmer Other documents probably use different fonts

**Comment by corkine** [comment]
I also encountered this problem. The documentation is numpy, scipy and qt5. 
Sometimes Zeal does not work, My Windows version is Windows 8.1, Chinese Version but I use English as primary display language . Zeal x64 0.5.0 . OpenURL always works.

![errorinfo1](https://user-images.githubusercontent.com/24912203/36632077-9c1714fa-19bc-11e8-9208-9155bcf77496.png)

![errorinfo2](https://user-images.githubusercontent.com/24912203/36632081-a6c65406-19bc-11e8-98c5-5ba202c571ee.png)

**Comment by fihyer** [comment]
I'm running **Zeal version 0.6.1** on **Windows 10 pro x64**, with both **region** and **system locale** set to **US**, and **English** as **display language**. Still facing the similar problem described above.
![image](https://user-images.githubusercontent.com/19499984/48301202-e6e2b280-e524-11e8-81ab-e28c4e4544c8.png)
Please help

**Comment by magicdmer** [comment]
version:0.6.1, partial wrong text 
![](https://images2.imgbox.com/38/5b/mihKcUBE_o.png)

**Comment by corkine** [comment]
Zeal 0.6.1 on Windows 8， Chinese Version but use English Language as display，both Java SE、Scala、JavaFx all have same issue, but not include same JavaDoc tech page like Java EE, Spring Framework.

**Comment by dsafa** [comment]
I also have this problem with `Zeal 0.6.1` on Windows 10 with language set to English.
Not sure about everyone else here but the problem only occurs when I have [mac type](https://github.com/snowie2000/mactype) enabled. If I disable it everything renders normally. Hopefully that helps.

**Comment by corkine** [comment]
> I also have this problem with `Zeal 0.6.1` on Windows 10 with language set to English.
> Not sure about everyone else here but the problem only occurs when I have [mac type](https://github.com/snowie2000/mactype) enabled. If I disable it everything renders normally. Hopefully that helps.

This is the case, I have ejected Mactype for Zeal rendering, and then Java, Qt, Scala documents can be displayed properly, cool! Even Chinese systems can display normally. I think that Oriental text may not work well on Windows, so many people use Mactype to optimize fonts. I didn't expect it to be a Mactype problem!

**Comment by magicdmer** [comment]
> I also have this problem with `Zeal 0.6.1` on Windows 10 with language set to English.
> Not sure about everyone else here but the problem only occurs when I have [mac type](https://github.com/snowie2000/mactype) enabled. If I disable it everything renders normally. Hopefully that helps.
Thank you, i add zeal.exe to mactype's exclusion list , zeal works ok now.  
But it's interesting that the first time i meet this problem , i disable mactype for test, but the problem not solved , so maybe i need test it for some time

**Comment by magicdmer** [comment]
@trollixx  the issue maybe can close,  new version zeal works well without mactype

**Comment by trollixx** [comment]
Thanks for the update! I didn't know about MacType. Is it popular among users with CJK system language?

**Comment by magicdmer** [comment]
> Thanks for the update! I didn't know about MacType. Is it popular among users with CJK system language?
MacType is a opensource font rendering tool for Windows, you can find it on github.  In fact ， it's popular among users who want to make their system font more perfect in the world.

**Comment by annulen** [comment]
Apple's way of rendering fonts is... peculiar, to say mildly. I don't think it's supported by Qt in any way

**Comment by annulen** [comment]
https://github.com/snowie2000/mactype/issues/195

**Comment by trollixx** [comment]
@annulen the ticket says there's also a known issue with some Windows API that QtWK calls, maybe that's what causing our other font rendering issues on Windows?

**Comment by annulen** [comment]
QtWebKit doesn't use Windows API for font manipulations, everything is done via QRawFont and QFontDatabase. So bugs should be reproducible in regular Qt widgets
